arwen/host/execution.go

```diff
@@ -289,13 +291,13 @@ func (host *vmHost) ExecuteOnSameContext(input *vmcommon.ContractCallInput) erro
 		return err
 	}
 
-	err = host.Async().Execute()
+	// err = host.Async().Execute()
```

Comment:
why is this commented ?

Reply:
It should have been deleted instead. `ExecuteOnSameContext()` should not attempt to execute any `AsyncCalls` on its own. That is the responsibility of the contract that calls `executeOnSameContext()` instead, as opposed to `executeOnDestContext()` which must complete everything before terminating.

In the next PR, the `AsyncContext` might be locked completely when entering `ExecuteOnSameContext()`, so that no new `AsyncCalls` are added to it.

Reply:
that would be nice - maybe we could even error if there is a call from executeonsamecontext - and makes an asynccall.